docs/puzzles/day17.md

```diff
@@ -2,10 +2,436 @@ import Solver from "../../../../website/src/components/Solver.js"
 
 # Day 17: Trick Shot
 
+by [@bishabosha](https://twitter.com/bishabosha)
+
 ## Puzzle description
 
 https://adventofcode.com/2021/day/17
 
+## Solution of Part 1
+
+### Modelling The Domain
+
+#### A Moving Probe
+
+The problem asks us to consider the trajectory of a projectile _probe_
+which has both a _position_ and a _velocity_. Both positions and velocities
+have two directions, `x` and `y`, with integer values. We model them
+with case classes:
+
+```scala
+case class Position(x: Int, y: Int)
+```
+```scala
+case class Velocity(x: Int, y: Int)
+```
+
+and then the probe is as follows:
+
+```scala
+case class Probe(position: Position, velocity: Velocity)
+```
+
+We also find out that a probe always has an initial position of `0,0`, so
+we model that also:
+
+```scala
+val initial = Position(x = 0, y = 0)
+```
+
+We are also told about the projectile motion of the probe - it moves
+in discrete steps, which we implement as follows:
+- on each step we create a new probe with a new position and a new velocity,
+- the `x` of the new position is the sum of `x` of the old position and `x`
+  of the old velocity,
+- the `y` of the new position is the sum of `y` of the old position and `y`
+  of the old velocity,
+- the `x` of the new velocity is given by subtracting the sign (`-1`, `0`
+  or `1`) of the `x` of the old velocity from itself (drag slows down the
+  projectile),
+- the `y` of the new velocity is given by subtracting `1` from the `y` of
+  the old velocity (due to gravity).
+
+The code is written as such:
+```scala
+def step(probe: Probe): Probe =
+  val Probe(pos, vel) = probe
+  Probe(
+    Position(x = pos.x + vel.x, y = pos.y + vel.y),
+    Velocity(x = vel.x - vel.x.sign, y = vel.y - 1)
+  )
+```
+
+#### A Target Area
+
+Next we are told that a successful launch trajectory causes the probe
+to be within in a target area after at least one of its steps. The area is
```

Comment:
Is this really correct to say “within in”?

Reply:
Ah no, it should be "within a target", indeed. ;)

Reply:
good spot, I fixed it